### PR TITLE
#6 Add preconditions and JSR-305 annotations to Instrument class

### PR DESCRIPTION
**Pull request by salvadore-jefferson**
New import statements to Instrument.java:
 *import static com.google.common.base.Preconditions.checkNotNull;
 *import javax.annotation.Nonnull;
 *import javax.annotation.Nullable;

Refactoring:
 *Changed instrumentNumber to instrumentKey
 *Changed String name to final String instrumentName
 *Changed int number to final int instrumentKey
 *Changed getNumber() to get Key()

*Added javadoc comments to the Instrument constructor
*Added JSR-305 annotations to the Instrument constructor
*Added preconditions to the Instrument constructor

*Added Google Guava dependency to the POM

*Ran a test with “null” as the Instrument name. The application failed
to compile and printed out my custom error message for that
precondition.

Note: This seems to be working, but I am not 100% convinced that it is
exactly the right way I should be doing this.

### CODE REVIEW

**Comment by salvadore-jefferson** [comment]
New pull request for branch BEATBOX-issue/6

File: src/main/java/com/github/salvadore_jefferson/beatbox/Instrument.java [hunk 1]
```diff
@@ -11,12 +16,20 @@
...
-    this.instrumentNumber = number;
+  /**
+   * Instrument constructor for creating MIDI Instruments. Requirements:
+   * 
+   * @param instrumentName
+   * @param instrumentKey
+   */
+  public @Nullable Instrument(@Nonnull final String instrumentName,
+      @Nonnull final int instrumentKey) {
+    checkNotNull(instrumentName, "You must provide a name for this Instrument");
```
Comment (garretwilson) [inline src/main/java/com/github/salvadore_jefferson/beatbox/Instrument.java]: Is this compiling? How is Java finding the Guava `checkNotNull()` method when you haven't included Guava in your Maven dependencies?

**Comment by salvadore-jefferson** [comment]
Yes it compiles. I downloaded the Google Guava jar file form here
https://code.google.com/p/guava-libraries/
And added it to the build path. Everything seems to be working, but like I said, I am not sure if I have it all configured in the best possible way.

**Comment by salvadore-jefferson** [comment]
I see now I miss labeled my commit message.

I said I added Google Guava to the POM, but what I added was Google findbugs for JSR-305

**Comment by garretwilson** [comment]
> I downloaded the Google Guava jar file form here ...
> And added it to the build path.

Think about this: if you add it to the build path, how are other developers going to get it? Remember, we took the Eclipse project files out of the repository. In addition, if you were ever to tell Eclipse to refresh the project from Maven, it would remove this from the build path, as Maven knows nothing (yet) about Guava.

Review the "dependency management" function of Maven. If Maven can go out and download our dependencies automatically, just based on their coordinates, why would we want to go download them manually?

In the documentation I gave you, when you see "dependency" think "Maven POM `<dependency>`". Go to the main Google Guava page (linked in the documentation). It should show you exactly what should go in your POM. Maven will handle downloading it automatically, just as it does for your JSR-305 dependency.

**Comment by garretwilson** [comment]
Note that the branch requested was `issues/6`, not `BEATBOX-issue/6`. The reason I had mentioned `BEATBOX-...` originally is because this is the format used by JIRA. But as we're switching to the GitHub issue system, we don't need (or want) the `BEATBOX-` part. Additionally, we want to use the format `issues/6` rather than `issue/6`. This is sort of nit-picky, but this actually goes to a larger discussion on RESTful representations and the semantics of a directory. There's no completely "correct" answer here, but there is much to be said for naming a collection resource in the plural. But that's a discussion for another day; just use the format `issues/6` for future branches.

**Comment by salvadore-jefferson** [comment]
> Note that the branch requested was issues/6
Noted Garret. The next branch I create will be branch `issues/7`

**Pull request merged**